### DISCUSSION TITLE
Hover state activates on sibling elements

### DISCUSSION THREAD

**Discussion by f3bw** [tailwindlabs/tailwindcss#11780] [Help] [answered]
Hello! So far I have not succeeded with this. But I want my navbar to have to following styling functionality. When on active page, the text of the active link is dark ( already have that part ). But I want to add a hover style to the other sibling links. For testing instance, I want all the links I am NOT hovering on, to be red. And the one I am hovering on, to be blue.

This is my code set up with nextJS

Navigation component:
```
import Link from 'next/link'
import { Navigation } from '../../templates/Navigation/router'

const navLinks = [
  { name: 'Projects', href: '/projects' },
  { name: 'CV', href: '/cv' },
  { name: 'About me', href: '/about' },
  { name: 'Contact', href: '/contact' },
]

const NavigationBar = () => {
  return (
    <>
      <div className='h-20'></div>
      <div className='fixed left-1/2 top-10 z-[5000] w-full max-w-[92%] -translate-x-1/2 rounded-xl bg-[rgba(220,220,220,.4)] backdrop-blur-lg'>
        <div className='flex items-center justify-between px-5 py-2.5'>
          <Link href={'/'}>Title</Link>
          <nav className='flex gap-4'>
            <Navigation navLinks={navLinks} />
          </nav>
        </div>
      </div>
    </>
  )
}
export default NavigationBar
```

Navlinks component:
```
'use client'

// nextjs-router.js
import { usePathname } from 'next/navigation'
import Link from 'next/link'

export function Navigation({ navLinks }) {
  const pathname = usePathname()

  return (
    <>
      {navLinks.map((link) => {
        const isActive = pathname.startsWith(link.href)

        return (
          <Link
            className={
              isActive
                ? 'font-medium text-black '
                : 'cursor-pointer font-medium text-black/40 transition duration-150 hover:text-black'
            }
            href={link.href}
            key={link.name}
          >
            {link.name}
          </Link>
        )
      })}
    </>
  )
}

```

Could someone help me with this or steer me in the right direction? Cheers!

**Comment by wongjn** [comment]
For not-hovered styling, have the text color applied with no variant. Change `text-black/40` to `text-red-500`. For hover color, change `hover:text-black` to `hover:text-blue-500`:

```diff
 className={
   isActive
     ? 'font-medium text-black '
-    : 'cursor-pointer font-medium text-black/40 transition duration-150 hover:text-black'
+    : 'cursor-pointer font-medium text-red-500 transition duration-150 hover:text-blue-500'
 }
 href={link.href}
```
See [this example in Tailwind Play](https://play.tailwindcss.com/RaqhMTn8vt).

**Reply by Semako123** [reply]
I also think this will work.

**Reply by f3bw** [reply]
Thanks this is indeed a step of the way. But the behaviour I want (using this current code as an example), is want to blue color to change to green, when I hover one of the links. So the one I am actually hovering on turn red and at that moment all of the sibling elements turn green. So blue as a base colour, red for the element that is being hovered and green for when one of the siblings is being hovered but not itself.

Does that make sense?

**Reply by wongjn** [reply, marked as answer]
Thank you for the clarification. You could consider using `group-hover` to turn the links green when the wrapper for the links is hovered. Since it has `gap` applied, we'll need to use `pointer-events: none` on the wrapper and then `pointer-events: auto` so that when hovering the gap, the links do not turn green:

[Tailwind Play example](https://play.tailwindcss.com/fBwQBgjuwy)

```diff
   <Link href={'/'}>Title</Link>
-  <nav className='flex gap-4'>
+  <nav className='flex gap-4 group pointer-events-none'>
     <Navigation navLinks={navLinks} />
   </nav>
 </div>
```
```diff
 className={
   isActive
-    ? 'font-medium text-black'
-    : 'cursor-pointer font-medium text-red-500 transition duration-150 hover:text-blue-500'
+    ? 'font-medium text-black pointer-events-auto'
+    : 'cursor-pointer font-medium text-red-500 transition duration-150 pointer-events-auto hover:!text-blue-500 group-hover:text-green-500'
 }
 href={link.href}
```

**Reply by f3bw** [reply]
This is exactly the behaviour I was looking for. Much appreciated! 

I also want to give a thank you for being so active in this community. I always see that you answer a lot of questions and I think that is really nice.
